### PR TITLE
Update cosmos-sdk to v0.47.7

### PR DESCRIPTION
**Pull request by DracoLi**
## Description
See `Kava Cosmos v47.7 Upgrade Changes` doc.

## Checklist
 - [x] Changelog has been updated as necessary.

### CODE REVIEW

**Review by galxy25 — APPROVED** [review]
all tests pass and app builds with rocksdb, approved for protonet deploy with async code and security audit ongoing as part of validation process before public testnet launch

File: app/app.go [hunk 1]
```diff
@@ -193,6 +196,7 @@ var (
 		crisis.AppModuleBasic{},
 		slashing.AppModuleBasic{},
 		ibc.AppModuleBasic{},
+		ibctm.AppModuleBasic{},
```
Comment (nddeluca) [inline app/app.go:199]: https://github.com/cosmos/ibc-go/pull/2825/files, this just registers the codec's as it's not in ibc core anymore

File: app/app_test.go [hunk 2]
```diff
@@ -143,3 +146,13 @@ func unmarshalJSONKeys(jsonBytes []byte) ([]string, error) {
 
 	return keys, nil
 }
+
+func removeIbcTmModule(modules []string) []string {
+	var result []string
+	for _, str := range modules {
+		if str != "07-tendermint" {
```
Comment (nddeluca) [inline app/app_test.go:153]: We can use `"github.com/cosmos/ibc-go/v7/modules/light-clients/07-tendermint".ModuleName` here

File: cmd/kava/cmd/app.go [hunk 3]
```diff
@@ -62,7 +64,7 @@ func (ac appCreator) newApp(
 
 	homeDir := cast.ToString(appOpts.Get(flags.FlagHome))
 	snapshotDir := filepath.Join(homeDir, "data", "snapshots") // TODO can these directory names be imported from somewhere?
-	snapshotDB, err := sdk.NewLevelDB("metadata", snapshotDir)
+	snapshotDB, err := cometbftdb.NewGoLevelDB("metadata", snapshotDir)
```
Comment (nddeluca) [inline cmd/kava/cmd/app.go:67]: This is a regression here -- should be `cometbftdb.NewDB("metadata", server.GetAppDBBackedn(appOpts), snapshotDir)` to open using configured db type

File: cmd/kava/cmd/app.go [hunk 4]
```diff
@@ -89,6 +91,24 @@ func (ac appCreator) newApp(
 		cast.ToUint32(appOpts.Get(server.FlagStateSyncSnapshotKeepRecent)),
 	)
 
+	// Setup chainId
+	home := cast.ToString(appOpts.Get(flags.FlagHome))
+	chainID := cast.ToString(appOpts.Get(flags.FlagChainID))
+	if len(chainID) == 0 {
```
Comment (nddeluca) [inline cmd/kava/cmd/app.go:97]: I believe this should fallback to genesis.json and not the client.toml

File: go.mod [hunk 5]
```diff
@@ -1,36 +1,38 @@
...
+	github.com/cosmos/cosmos-sdk v0.47.7
 	github.com/cosmos/go-bip39 v1.0.0
-	github.com/cosmos/ibc-go/v6 v6.1.1
+	github.com/cosmos/gogoproto v1.4.10
+	github.com/cosmos/ibc-go/v7 v7.3.1
 	github.com/ethereum/go-ethereum v1.10.26
 	github.com/evmos/ethermint v0.21.0
 	github.com/go-kit/kit v0.12.0
-	github.com/gogo/protobuf v1.3.3
+	github.com/gogo/protobuf v1.3.2
```
Comment (nddeluca) [inline go.mod:20]: Downgrade?

File: x/validator-vesting/querier.go [hunk 6]
```diff
@@ -1,358 +0,0 @@
-package validator_vesting
```
Comment (nddeluca) [inline x/validator-vesting/querier.go:1]: Just noting that I believe coinmarketcap does use these.  I'm not sure a grpc querier would let us return an int value as the response

**Review by nddeluca — APPROVED** [review]
Looks good.  I think in a follow on PR we should switch all of our Dec usage to sdkmath instead of the legacy types, but not required for this PR

**Review by Zgeordo — APPROVED** [review]

**Pull request merged**